pkg/cluster/cluster.go

```diff
@@ -118,12 +117,16 @@ func (c *Cluster) setup() error {
 		return fmt.Errorf("invalid cluster spec: %v", err)
 	}
 
-	var phase spec.ClusterPhase
+	// Set status here because if the below state check fails, it will use the status field
```

Comment:
We need to cleanup the comment. I understand what we want to do here, but even if then I do not understand the comment.